### PR TITLE
[in_app_purchase] Add api to expose country code

### PR DESCRIPTION
**Pull request by reidbaker**
- **Add changelog and version bump**
- **dependency override from package tool**

Fixes flutter/flutter/issues/141627

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by LouiseHsu — APPROVED** [review]
ios side lgtm! thanks for adding the ios implementation too!

File: packages/in_app_purchase/in_app_purchase_platform_interface/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
-## NEXT
+## 1.4.0
 
+* Adds getCountryCode api.
```
Comment (gmackall) [inline packages/in_app_purchase/in_app_purchase_platform_interface/CHANGELOG.md:3]: nit:
```suggestion
* Adds getCountryCode API. 
```

File: packages/in_app_purchase/in_app_purchase_platform_interface/lib/src/in_app_purchase_platform.dart [hunk 2]
```diff
@@ -194,4 +194,21 @@ abstract class InAppPurchasePlatform extends PlatformInterface {
   ///    [PurchaseDetails.verificationData].
   Future<void> restorePurchases({String? applicationUserName}) =>
       throw UnimplementedError('restorePurchases() has not been implemented.');
+
+  /// Returns the users country.
```
Comment (gmackall) [inline packages/in_app_purchase/in_app_purchase_platform_interface/lib/src/in_app_purchase_platform.dart:198]: nit: 
```suggestion
  /// Returns the user's country.
```

**Review by gmackall — APPROVED** [review]
Android LGTM! Just some nitpicky docs comments

File: packages/in_app_purchase/in_app_purchase/lib/in_app_purchase.dart [hunk 3]
```diff
@@ -209,4 +209,21 @@ class InAppPurchase implements InAppPurchasePlatformAdditionProvider {
...
+  /// See: https://unicode.org/cldr/charts/latest/supplemental/territory_containment_un_m_49.html
+  ///
+  /// iOS:
+  /// Returns the country code from SKStoreFrontWrapper.
+  ///
+  /// See: https://developer.apple.com/documentation/storekit/skstorefront?language=objc
+  ///
+  ///
+  Future<void> countryCode() =>
+      throw UnimplementedError('countryCode() has not been implemented.');
```
Comment (stuartmorgan) [inline packages/in_app_purchase/in_app_purchase/lib/in_app_purchase.dart:228]: Shouldn't this be calling the new platform interface method?

File: packages/in_app_purchase/in_app_purchase/pubspec.yaml [hunk 4]
```diff
@@ -2,7 +2,7 @@ name: in_app_purchase
 description: A Flutter plugin for in-app purchases. Exposes APIs for making in-app purchases through the App Store and Google Play.
 repository: https://github.com/flutter/packages/tree/main/packages/in_app_purchase/in_app_purchase
 issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+in_app_purchase%22
-version: 3.1.13
+version: 3.2.0
 
 environment:
   sdk: ^3.1.0
```
Comment (stuartmorgan) [inline packages/in_app_purchase/in_app_purchase/pubspec.yaml:8]: FYI these values will need to be updated to match `i_a_p_storekit`'s lower limits when you do the final app-facing PR, otherwise `pub get` will fail.

File: packages/in_app_purchase/in_app_purchase/lib/in_app_purchase.dart (hunk 3 above)
Comment (stuartmorgan) [reply packages/in_app_purchase/in_app_purchase/lib/in_app_purchase.dart:228]: And you'll want to add Dart unit tests at this layer that it is in fact calling through.

File: packages/in_app_purchase/in_app_purchase_platform_interface/CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,6 @@
-## NEXT
+## 1.4.0
 
+* Adds getCountryCode API.
```
Comment (stuartmorgan) [inline packages/in_app_purchase/in_app_purchase_platform_interface/CHANGELOG.md:3]: Nit: ```Adds `getCountryCode` API.```

File: packages/in_app_purchase/in_app_purchase_platform_interface/lib/src/in_app_purchase_platform.dart [hunk 6]
```diff
@@ -194,4 +194,21 @@ abstract class InAppPurchasePlatform extends PlatformInterface {
...
+  /// See: https://unicode.org/cldr/charts/latest/supplemental/territory_containment_un_m_49.html
+  ///
+  /// iOS:
+  /// Returns the country code from SKStoreFrontWrapper.
+  ///
+  /// See: https://developer.apple.com/documentation/storekit/skstorefront?language=objc
+  ///
+  ///
+  Future<void> countryCode() =>
+      throw UnimplementedError('countryCode() has not been implemented.');
```
Comment (stuartmorgan) [inline packages/in_app_purchase/in_app_purchase_platform_interface/lib/src/in_app_purchase_platform.dart:213]: You'll need a test to land the stand-alone PR that gets split out of this one. Generally we just test that the default throws Unimplemented.

**Review by stuartmorgan — COMMENTED** [review]
Minor comments, but this looks good to split out the first PR (with just the platform interface part).

File: packages/in_app_purchase/in_app_purchase/lib/in_app_purchase.dart (hunk 3 above)
Comment (reidbaker) [reply packages/in_app_purchase/in_app_purchase/lib/in_app_purchase.dart]: Done! Thanks for the out of band explanation of this packages value.

File: packages/in_app_purchase/in_app_purchase_platform_interface/lib/src/in_app_purchase_platform.dart (hunk 6 above)
Comment (reidbaker) [reply packages/in_app_purchase/in_app_purchase_platform_interface/lib/src/in_app_purchase_platform.dart:213]: Thanks, added.

**Comment by reidbaker** [comment]
Assuming the tests pass this pr is ready to land.

**Review by LouiseHsu — APPROVED** [review]
LGTM!

File: packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md [hunk 7]
```diff
@@ -1,6 +1,6 @@
 ## 0.3.4
 
-* Adds `countryCode` API.
```
Comment (stuartmorgan) [inline packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md:3]: You'd need to bypass checks to fix the things in this package without a release; I would just revert them and we can clean them up as drive-bys later.

**Comment by reidbaker** [comment]
I gave the pr a version change exemption because I was fixing the formatting of the android package.

File: packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md (hunk 7 above)
Comment (stuartmorgan) [reply packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md:3]: Oh, you already bypassed the check. That works too.

**Review by stuartmorgan — APPROVED** [review]

File: packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md (hunk 7 above)
Comment (reidbaker) [reply packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md]: ah ok

Comment (reidbaker) [reply packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md]: I saw your comment right after. This version has no android changes and the version check override is removed.
